src/plugins/intel_cpu/src/emitters/x64/jit_snippets_emitters.cpp

```diff
@@ -856,48 +880,147 @@ void BrgemmEmitter::initBrgemm(brgemmCtx& ctx, std::unique_ptr<brgemm_kernel_t>&
+                Label K_loop_begin;
+                // Note: we never emit loop for the first blocked kernel, since it always executed only once.
+                // The purpose of the first blocked K kernel is to initializes output, because it has beta = 0
+                if (k_blocked_id == 0) {
+                    total_K_work_amount -= brgemmCtx.K;
+                } else if (m_K_blk_loop) {
+                    h->mov(work_amount_K, total_K_work_amount);
+                    h->L(K_loop_begin);
+                }
 
-        size_t brgIdx0 = getBrgIdx(0, 0);
-        size_t K0_step0 = m_brgCtxs0[brgIdx0].K;
-        size_t K0_step1 = m_brgCtxs0[brgIdx0].K * m_brgCtxs0[brgIdx0].LDB;
-        size_t N0_step0 = m_brgCtxs0[brgIdx0].N * m_brg0VnniFactor;
-        size_t N0_step1 = m_brgCtxs0[brgIdx0].N;
-        for (size_t n = 0; n < 2; n++) {
-            for (size_t k = 0; k < 2; k++) {
-                auto& brgemmCtx = m_brgCtxs0[getBrgIdx(k, n)];
-
-                if (brgemmCtx.K != 0 && brgemmCtx.N != 0) {
-                    const size_t in0_offset = m_load_offset_a + k * K0_step0 * io_data_size[0];
-                    const size_t in1_offset = m_load_offset_b + (k * K0_step1 + n * N0_step0) * io_data_size[1];
-                    const size_t in2_offset = m_load_offset_scratch + (m_with_comp ? n * N0_step1 * sizeof(int32_t) : 0);
-                    const size_t out0_offset = m_store_offset_c + n * N0_step1 * io_data_size[2];
-
-                    emit_brgemm_kernel_call(m_brgKernels0[getBrgIdx(k, n)].get(),
-                                            brgemmCtx,
-                                            input_0,
-                                            input_1,
-                                            input_2,
-                                            output_0,
-                                            in0_offset,
-                                            in1_offset,
-                                            in2_offset,
-                                            out0_offset);
+                emit_N_blocking_loops(k_blocked_id, input_0, input_1, input_2, output_0, work_amount_N);
+                h->add(input_0, brgemmCtx.K * io_data_size[0]);
+                h->add(input_1, (brgemmCtx.K * brgemmCtx.LDB) * io_data_size[1]);
+                if (m_K_blk_loop && k_blocked_id) {
+                    h->sub(work_amount_K, brgemmCtx.K);
+                    h->cmp(work_amount_K, brgemmCtx.K);
+                    h->jge(K_loop_begin);
                 }
             }
         }
+        // K loop tail
+        if (get_K_kernel_idx(k_tail_id, kernel_idx)) {
+            emit_N_blocking_loops(k_tail_id, input_0, input_1, input_2, output_0, work_amount_N);
+            h->sub(input_1, m_N * io_data_size[1]);
+            h->sub(output_0, m_N * io_data_size.back());
+            if (m_with_scratch && m_with_comp)
+                h->sub(input_2, m_N * io_data_size[2]);
```

Comment:
We already do it in `emit_N_blocking_loops`. How does it work? :expressionless:
Could you please add the tests to cover it as well? Thanks!

Reply:
It doesn't work actually. My bad
Thank you very much for this finding! The issue has been fixed and covered by tests